app/models/visitors_step.rb

```diff
@@ -79,9 +84,41 @@ def validate_email
     end
   end
 
+  # rubocop:disable Metrics/AbcSize
+  # rubocop:disable Metrics/MethodLength
```

Comment:
I must say I see no great virtue in splitting. It's a private method, so not testable in isolation anyway, and you're never going to call one half without the other. If you want to split it in two, I won't object, but I won't be bothering personally :)